src/library/autodj/autodjprocessor.cpp

```diff
@@ -723,19 +782,25 @@ bool AutoDJProcessor::removeTrackFromTopOfQueue(TrackPointer pTrack) {
     return true;
 }
 
-void AutoDJProcessor::playerPlayChanged(DeckAttributes* pAttributes, bool playing) {
+void AutoDJProcessor::playerPlayChanged(DeckAttributes* thisDeck, bool playing) {
     if (sDebug) {
-        qDebug() << this << "playerPlayChanged" << pAttributes->group << playing;
+        qDebug() << this << "playerPlayChanged" << thisDeck->group << playing;
+    }
+    // In case both decks were stopped and now this one just started, make this
+    // deck the "from deck".
+    if (playing && !getOtherDeck(thisDeck)->isPlaying()) {
+        calculateTransition(thisDeck, getOtherDeck(thisDeck), false);
     }
-    // We may want to do more than just calculate fade thresholds when playing
-    // state changes so keep these two as separate methods for now.
 
-    // This will calculate the Transition to the already loaded, in most cases
-    // already played other track.
-    // This is required because the user may have loaded a track or changed play
-    // manually
-    if (playing) {
-        calculateTransition(pAttributes, getOtherDeck(pAttributes));
+    // If the user manually pressed play on the "to deck" before fading, for
+    // example to adjust the intro/outro cues, and lets the deck play until the
+    // end, seek back to the start point instead of keeping the deck stopped at
+    // the end.
+    if (!playing && thisDeck->playPosition() == 1.0 && !thisDeck->isFromDeck) {
```

Comment:
> The from deck is stopped once it reaches the outro start point.

I do not understand. In what case does this happen?

Reply:
Start Auto Dj, seek the too deck to the very and and watch.

This can happen if a user whishes to set an outro point and forget about it.
Obviously a rare case, but I like the Auto DJ to deal with any situation.

Reply:
This bug still happens.

Reply:
I can't reproduce this. If I seek the "to deck" to the very end, then let the "from deck" play until it transitions, the "to deck"'s track gets skipped and the next one is loaded and played. AutoDJ does not stop. Skipping the track may be surprising, but it makes sense.